File: 003_largest_prime_factor.exs
# What is the largest prime factor of the number 600851475143 ?

defmodule Primes do

	def get_largest_prime(number, counter) do
		result =
			case counter do
				c when c <= number ->
					case rem(number, counter) do
						0 -> 
							number = div(number, counter)
							get_largest_prime(number, 2)
						_ ->
							get_largest_prime(number, counter+1)
					end
				_ ->
					counter
			end
		result
	end
end

IO.inspect(Primes.get_largest_prime(600851475143, 2), label: "--------------")
# 6857

File: 003_largest_prime_factor.py
# What is the largest prime factor of the number 600851475143 ?

def find_largest_prime(num):
	result = 0
	counter = 2
	while counter <= num:
		if num % counter == 0:
			if counter > result:
				result = counter
			num = num / counter
			counter = 2
		else:
			counter += 1
	return result

print(find_largest_prime(600851475143))

# 6857